src/dev/i18n/extract_default_translations.js

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { resolve } from 'path';
+
+import { extractHtmlMessages } from './extract_html_messages';
+import { extractCodeMessages } from './extract_code_messages';
+import { extractJadeMessages } from './extract_jade_messages';
+import {
+  globAsync,
+  makeDirAsync,
+  pathExists,
+  readFileAsync,
+  writeFileAsync,
+} from './utils';
+
+const ESCAPE_CHARACTERS_REGEX = /\\([\s\S])|(')/g;
+
+function addMessageToMap(targetMap, key, value) {
+  const existingValue = targetMap.get(key);
+  if (targetMap.has(key) && existingValue.message !== value.message) {
+    throw new Error(
+      `There is more than one default message for the same id "${key}": "${existingValue}" and "${value}"`
+    );
+  }
+  targetMap.set(key, value);
+}
+
+export async function extractDefaultTranslations(inputPath) {
+  const entries = await globAsync('*.{js,jsx,jade,ts,tsx,html}', {
+    cwd: inputPath,
+    matchBase: true,
+  });
+
+  const { htmlEntries, codeEntries, jadeEntries } = entries.reduce(
+    (paths, entry) => {
+      const resolvedPath = resolve(inputPath, entry);
+
+      if (resolvedPath.endsWith('.html')) {
+        paths.htmlEntries.push(resolvedPath);
+      } else if (resolvedPath.endsWith('.jade')) {
+        paths.jadeEntries.push(resolvedPath);
+      } else {
+        paths.codeEntries.push(resolvedPath);
+      }
+
+      return paths;
+    },
+    { htmlEntries: [], codeEntries: [], jadeEntries: [] }
+  );
+
+  const defaultMessagesMap = new Map();
+
+  await Promise.all(
+    [
+      [htmlEntries, extractHtmlMessages],
+      [codeEntries, extractCodeMessages],
+      [jadeEntries, extractJadeMessages],
+    ].map(async ([entries, extractFunction]) => {
+      // If some file contains an error, we cannot handle it until all files are read.
+      // TODO: move files reading to "extract*" async generators after async iterators implementation (for-await-of syntax)
```

Comment:
Before we merge this in, I think we should remove TODO comments by resolving them, or keep them while also providing a link to an issue. (Or else we'd lose track of this)